### PR TITLE
[Identity] Documentation updates

### PR DESCRIPTION
**Pull request by pvaneck**
This fleshes out WAM/broker related documentation for the identity libraries.

Ref: https://github.com/Azure/azure-sdk-for-python/issues/30163

### CODE REVIEW

File: sdk/identity/azure-identity/TROUBLESHOOTING.md [hunk 1]
```diff
@@ -296,15 +296,15 @@ Get-AzAccessToken -ResourceUrl "https://management.core.windows.net"
 
 ## Troubleshoot WAM+MSA login issues
```
Comment (joshfree) [inline sdk/identity/azure-identity/TROUBLESHOOTING.md:297]: Let's put the full names here with the abbreviations in parenthesis, so people searching keywords are more likely to find this document

Comment (joshfree) [reply sdk/identity/azure-identity/TROUBLESHOOTING.md:297]: Here's the blurb from wikipedia that seems excellent


A Microsoft account or MSA[[1]](https://en.wikipedia.org/wiki/Microsoft_account#cite_note-1) (previously known as Microsoft Passport,[[2]](https://en.wikipedia.org/wiki/Microsoft_account#cite_note-pressp-2) .NET Passport, and Windows Live ID) is a [single sign-on](https://en.wikipedia.org/wiki/Single_sign-on) personal [user account](https://en.wikipedia.org/wiki/User_(computing)) for [Microsoft](https://en.wikipedia.org/wiki/Microsoft) customers to [log in](https://en.wikipedia.org/wiki/Login) to consumer[[3]](https://en.wikipedia.org/wiki/Microsoft_account#cite_note-3)[[4]](https://en.wikipedia.org/wiki/Microsoft_account#cite_note-4) Microsoft services (like [Outlook.com](https://en.wikipedia.org/wiki/Outlook.com)), devices running on one of Microsoft's current [operating systems](https://en.wikipedia.org/wiki/Operating_system) (e.g. [Microsoft Windows](https://en.wikipedia.org/wiki/Microsoft_Windows) computers and tablets, [Xbox](https://en.wikipedia.org/wiki/Xbox) consoles), and Microsoft [application software](https://en.wikipedia.org/wiki/Application_software) (including [Visual Studio](https://en.wikipedia.org/wiki/Visual_Studio)).

**Review by joshfree — APPROVED** [review]

File: sdk/identity/azure-identity/README.md [hunk 2]
```diff
@@ -318,6 +318,10 @@ Token caching is a feature provided by the Azure Identity library that allows ap
 
 The Azure Identity library offers both in-memory and persistent disk caching. For more details, see the [token caching documentation](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/identity/azure-identity/TOKEN_CACHING.md).
 
+## Brokered authentication
+
+An authentication broker is an application that runs on a user’s machine that manages the authentication handshakes and token maintenance for connected accounts. Currently, only the Windows authentication broker, Web Account Manager (WAM), is supported through an extension package, [`azure-identity-broker`][azure_identity_broker]. For details on authenticating using WAM, see the [broker plugin documentation][azure_identity_broker_readme].
```
Comment (billwert) [inline sdk/identity/azure-identity/README.md:323]:
```suggestion
An authentication broker is an application that runs on a user’s machine that manages the authentication handshakes and token maintenance for connected accounts. Currenly only the Windows Web Account Manager (WAM) is supported. To enable support include the [`azure-identity-broker`][azure_identity_broker] package. For details on authenticating using WAM, see the [broker plugin documentation][azure_identity_broker_readme].
```

File: sdk/identity/azure-identity-broker/README.md [hunk 3]
```diff
@@ -2,37 +2,73 @@
...
+
+- `azure.identity.broker.InteractiveBrowserBrokerCredential`
+- `azure.identity.broker.UsernamePasswordBrokerCredential`
+
+These credentials are subclasses of the `InteractiveBrowserCredential` and `UsernamePasswordCredential`
+classes of the Azure Identity library.
+
+### Parent window handles
+
+When authenticating interactively via `InteractiveBrowserBrokerCredential`, a parent window handle is required to ensure that the authentication dialog is shown correctly over the requesting window. In the context of graphical user interfaces on devices, a window handle is a unique identifier that the operating system assigns to each window. For the Windows operating system, this is an integer value that serves as a reference to a specific window.
```
Comment (pvaneck) [inline sdk/identity/azure-identity-broker/README.md:33]: MSA Passthrough section (legacy)

**Review by scottaddie — APPROVED** [review]
Nit: Change the https://azure.microsoft.com/free/ links to https://azure.microsoft.com/free/python.

File: sdk/identity/azure-identity/TROUBLESHOOTING.md (hunk 1 above)
Comment (pvaneck) [reply sdk/identity/azure-identity/TROUBLESHOOTING.md]: Updated

**Comment by pvaneck** [comment]
Thanks, all, for the feedback. Will merge this as I want to get the azure-identity readme doc updated for the Python GA tomorrow.

I will add a tidbit about the redirect URI configuration in the broker README in a follow-up PR.

File: sdk/identity/azure-identity/TROUBLESHOOTING.md [hunk 4]
```diff
@@ -294,17 +294,17 @@ Get-AzAccessToken -ResourceUrl "https://management.core.windows.net"
...
 
 ![MSA Microsoft Entra only](./images/MSA1.png)
 
-If you choose "Use another account" and type in an MSA outlook.com account, it fails:
+If you choose "Use another account", and type in an MSA outlook.com account, it fails:
 
 ![Fail on use another account](./images/MSA2.png)
 
-Since azure-identity 1.15.0b1, you can set `enable_msa_passthrough`` to `True`, and MSA outlook.com accounts that are logged in to Windows are automatically listed:
+When constructing the credential, you can set the `enable_msa_passthrough` keyword argument to `True`, and MSA outlook.com accounts that are logged in to Windows are automatically listed:
```
Comment (KarishmaGhiya) [inline sdk/identity/azure-identity/TROUBLESHOOTING.md:307]: Should we add in error codes and error names for `NativeAuthError` so users will know it's from the broker?

**Pull request merged**